docs/extend/extend-apiml/api-mediation-infinispan.md

```diff
@@ -0,0 +1,55 @@
+# Using Infinispan as a storage solution through the Caching service
+
+As an API developer, you can configure Infinispan as a storage solution through the Caching service. This article describes how to configure your storage solution for Infinispan.
+You can configure Infinispan for high availability as well as to replicate data to provide data durability and availability.
+
+- [Understanding Infinispan](#understanding-infinispan)
+- [Infinispan configuration](#infinispan-configuration)
+
+## Understanding Infinispan
+
+Infinispan is a storage solution that stores data structures in key-value pairs. The API Caching service uses hash sets, where each
+service storing data via the Caching service has its own hash, and each data entry is a key-value entry within that service's Infinispan hash set.
+
+For more information on Infinispan, see the [official Infinispan documentation](https://infinispan.org/documentation/).
+
+### Infinispan replica instances
+
+Infinispan can be used with one standalone instance. For data durability, however, a Cross-Site Replication configuration is recommended.
+Cross-site replication guarantees service continuity in the event of outages or disasters and presents client applications with a single point of access to data in globally distributed caches.
+
+For more information on Infinispan replication and how to configure a replica instance, see the [official Infinispan Cross-site Replication documentation](https://infinispan.org/docs/stable/titles/xsite/xsite.html).
+
+### Infinispan SSL/TLS
+
+Infinispan supports SSL/TLS. For information on enabled SSL/TLS with Infinispan, see the [official Infinispan Security documentation](https://infinispan.org/docs/stable/titles/security/security.html#secure-cluster-transport).
+
+### Infinispan and JGroups
+
+Infinispan uses [JGroups](http://www.jgroups.org/) as its underlying clustering layer. In order to configure the finer details of clustering (discovery, flow control, cross-site, etc) you have to provide a separate XML file with the desired configuration and reference this XML file from your Infinispan XML file as follows:
```

Comment:
we should not be telling users to create a new xml configuration file, we have done this on their behalf.

Reply:
yep my mistake, I wanted to briefly mention jgroups but eventually copied all the sentence by mistake. Probably it's not worthy to even mention it. I will remove this section